Assets/FunnyJuice/Scripts/PlayerInteractor.cs

```diff
@@ -0,0 +1,39 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerInteractionRay : MonoBehaviour
+{
+    [SerializeField] private LayerMask interactionMask;
+
+    private GameObject previousObjectHit;
+    
+    void Update()
+    {
+        if (!Physics.Raycast(transform.position, transform.forward, out RaycastHit hit, 10f, interactionMask))
+        {
+            if(previousObjectHit != null && previousObjectHit.TryGetComponent(out FunnyJuice previousFunnyJuice))
+                previousFunnyJuice.Unhighlight();
+            
+            return;
+        }
+
+        if (hit.collider.TryGetComponent(out FunnyJuice funnyJuice))
```

Comment:
This is where I can change it into a more general interaction system using an interface `IObjectInteractable`. If we have another object to interact with then I'll change that code but for now I think we are good 👍